Ilia_Prokofev

blowushka opened the pull request:
Completed homework_1

blowushka committed:
Change "Hello World" to "Hello Russia"

blowushka committed:
Change "Hello, Russia" to "Hello, Epam"

blowushka committed:
.gitignore update

blowushka committed:
Add new folder called homework_1 and complete homework

blowushka committed:
Edit args output a bit

blowushka committed:
Change russian words to english

blowushka committed:
Update .gitignore

blowushka committed:
Add if(args.length == 0) validation

blowushka committed:
Update .gitignore

blowushka committed:
Delete unnecessary files

blowushka committed:
Merge remote-tracking branch 'origin/master' into feature/IliaProkofev

# Conflicts:
#	.gitignore
#	src/main/java/homework_1/Main.java

blowushka committed:
Add README.md and update .gitignore

blowushka committed:
Update README.md

blowushka committed:
Delete build.gradle

blowushka committed:
Delete Main.java from wrong folder

blowushka committed:
Ignore .gitignore for push

blowushka committed:
Restore .gitignore to remote repo

blowushka committed:
Delete .gitignore from remote repo

blowushka committed:
Merge remote-tracking branch 'origin/feature/IliaProkofev' into feature/IliaProkofev

# Conflicts:
#	.gitignore

NikolaevArtem commented on src/main/java/homework_1/CharCount.java:
```diff
@@ -0,0 +1,25 @@
+            } else {
+                System.out.println(arg + ": " + arg.length());
+            }
+        }
```
Indentation looks broken

NikolaevArtem commented on src/main/java/homework_1/CharCount.java:
```diff
@@ -0,0 +1,25 @@
+        System.out.println("Count of Char's in words:");
+        for (String arg : args) {
+            if (arg.equals("error")) {
+                System.out.println(RED_BOLD + "Alarm!");
```
You need to have close escape string, otherwise all other text will be red color too

NikolaevArtem commented on .gitignore:
```diff
@@ -6,6 +6,7 @@
 /gradle/
 /.DS_Store/
 
+
```
Please add back deleted files and remove all unnecessary lines and so on

blowushka committed:
Restore .gitignore to remote repo

blowushka committed:
Fix indentation and RED-text output

NikolaevArtem commented on .gitignore:
```diff
@@ -34,4 +34,8 @@ hs_err_pid*
-.idea
\ No newline at end of file
+.idea
+/gradlew
```
These files should be included, please revert. Otherwise, I see 8 changed files instead of two

NikolaevArtem commented on README.md:
```diff
@@ -1,9 +1,9 @@
+| Index | Solution  | Short description
 | --- | --- | --- |
-| HW1 | [Console printer](https://github.com/NikolaevArtem/Java_Core_June_2021/tree/master/src/main/java/homework_1) | The app that reads input arguments and prints them, until "error" argument |
+| HW1 | [Console printer](https://github.com/NikolaevArtem/Java_Core_June_2021/tree/feature/IliaProkofev/src/main/java/homework_1) | The app that reads input arguments and prints them, until "error" argument |
```
It still looks at my branch

blowushka replied:
I've no change there anything, and it's looks at my branch _"tree/feature/IliaProkofev/src/main/java/homework_1"_ instead of  _"tree/master/src/main/java/homework_1"_ am I right?

NikolaevArtem replied:
You're right, sorry, I looked at wrong place

blowushka committed:
Restore deleted files

blowushka committed:
Fix indentation

NikolaevArtem commented on src/main/java/homework_1/CharCount.java:
```diff
@@ -0,0 +1,19 @@
+            System.out.println("Count of Char's in words:");
+            for (String arg : args) {
+                if (arg.equals("error")) {
+                    System.err.println("Alarm!");
```
Doesn't work as expected
![image](https://user-images.githubusercontent.com/58872509/125207756-af5cd300-e296-11eb-99c1-edfedcdd4c0f.png)

NikolaevArtem replied:
Still the same

blowushka committed:
Refactoring

blowushka committed:
Refactoring

blowushka committed:
Final refactoring

blowushka committed:
Complete Homework_2

blowushka committed:
Edit README.md

blowushka committed:
Delete unnecessary import

blowushka committed:
Change output for PyramidPrinter

blowushka committed:
Change program exit style

NikolaevArtem commented on src/main/java/homework_2/pyramid_printer/PyramidPrinter.java:
```diff
@@ -0,0 +1,18 @@
+
+import java.util.Scanner;
+
+public class PyramidPrinter {
```
Good

NikolaevArtem commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,61 @@
+import java.util.Random;
+import java.util.Scanner;
+
+public class RandomCharsTable {
```
Opt: exception handling
![image](https://user-images.githubusercontent.com/58872509/126038035-ce2fcc67-64a4-42d5-bddd-bef8c613e223.png)

NikolaevArtem commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,61 @@
+import java.util.Scanner;
+
+public class RandomCharsTable {
+    public static Scanner scanner = new Scanner(System.in);
```
Opt: why is it public?

NikolaevArtem commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,61 @@
+            checkStratAndExecute(strategy, table);
+        } catch (InputMismatchException e) {
+            System.out.println("You entered invalid value, try again.");
+            System.exit(0);
```
Bad way of exiting program. Also it seems you have memory leak, right? Where do you close scanner?

NikolaevArtem commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,117 @@
+package homework_2.traffic_light;
+
+import java.util.Arrays;
```
Opt: unused import

NikolaevArtem commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,117 @@
+            getSeconds(globalseconds);
+
+        } catch (InputMismatchException e) {
+            System.out.println("You are input incorrect date, try again");
```
Please check english writing

blowushka committed:
Fix minor problems, close Scanner in RandomCharsTable.java, fix English spelling.

blowushka committed:
Changed the entry point to applications, major refactoring

blowushka committed:
Merge branch 'master' into feature/IliaProkofev

blowushka committed:
Global refactoring:
--Extract frequently used methods and constants into separate modules
--Create tests for PyramidPrinter and RandomCharsTable
--Major and minor bug fixes

blowushka committed:
Complete all tests for homework, create Immutable Class, minor improvements and bug fixes.

blowushka committed:
Update Readme.md

blowushka committed:
Edit ImmutableClass and add test

NikolaevArtem commented on src/main/java/homework_2/pyramid_printer/PyramidPrinterD.java:
```diff
@@ -0,0 +1,27 @@
+import static homework_2.pyramid_printer.Utils.*;
+
+public class PyramidPrinterD {
+    protected static void run() {
```
You need to have a minimum of one method as public, otherwise, how to run/test it?

NikolaevArtem commented:
Please name classes according to homework chat, and use run() method. Method run() should be public

blowushka committed:
--Changed the class names as specified in the condition
--Made all run() methods public

blowushka committed:
Fixed typo in README.md

blowushka committed:
Change the type of abstraction

pepya4ka commented on src/main/java/homework_2/pyramid_printer/Main.java:
```diff
@@ -0,0 +1,13 @@
+package homework_2.pyramid_printer;
+
+public class Main {
+    public static void main(String[] args) {
```
```suggestion
public class Main {

    public static void main(String[] args) {
```

pepya4ka commented on src/main/java/homework_2/pyramid_printer/PyramidPrinter.java:
```diff
@@ -0,0 +1,28 @@
+import static homework_2.pyramid_printer.Utils.*;
+
+public class PyramidPrinter {
+    public void run() {
```
try to split this method

pepya4ka commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    private static void checkLengthAndWidth(int length, int width) {
+        if (length == 0 || width == 0){
```
```suggestion
        if (length <= 1 || width <= 1) {
```
covers several cases

pepya4ka commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,78 @@
+
+        for (int i = 0; i < length; i++) {
+            for (int j = 0; j < width; j++) {
+                table[i][j] = (char) (random.nextInt((90 - 65) + 1) + 65);
```
```suggestion
                table[i][j] = (char) (random.nextInt((26) + 65);
```

pepya4ka commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,78 @@
+        } catch (NegativeArraySizeException e) {
+            System.out.println("Array size cannot be negative.");
+        } catch (InputMismatchException e) {
+            System.out.println("Input parameters must be in the format [positive integer(>0)] [positive integer(>0)] [even|odd]");
```
check the output format in the task

pepya4ka commented on src/main/java/homework_2/traffic_light/ExtraTrafficLight.java:
```diff
@@ -0,0 +1,55 @@
+import static homework_2.traffic_light.ColorSelector.*;
+
+
+public class ExtraTrafficLight {
```
try extend your TrafficLight class

pepya4ka commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,56 @@
+                getSeconds(seconds);
+            }
+        } catch (InputMismatchException e) {
+            System.out.println("Only numbers is allow.");
```
check the output format in the task

pepya4ka commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,56 @@
+
+import static homework_2.traffic_light.ColorSelector.*;
+
+class TrafficLight {
```
try to split this method

pepya4ka commented on src/main/java/homework_3/ImmutableClassMan.java:
```diff
@@ -0,0 +1,61 @@
+    public ImmutableClassMan(int age) {
+        this.age = age;
+        this.name = "";
+        this.iq = new Integer(null);
```
```suggestion
        this.iq = null;
```
if i try to get 'iq' i can catch `NullPointerException`

pepya4ka commented on src/main/java/homework_3/ImmutableClassMan.java:
```diff
@@ -0,0 +1,61 @@
+    public ImmutableClassMan(int age, String name) {
+        this.age = age;
+        this.name = name;
+        this.iq = new Integer(null);
```
if i try to do this 
`ImmutableClassMan immutableClassMan = new ImmutableClassMan(24);`
i always catch `NumberFormatException`

blowushka committed:
Changed the names of the tests, added some blank lines, deleted tests for Immutable Class

blowushka committed:
Refactoring + README edit

blowushka committed:
README edit

blowushka committed:
Change .txt input

blowushka committed:
Refactoring

blowushka committed:
Add CustomAnnotation, edit README.md

blowushka committed:
Added test for CustomAnnotation

blowushka committed:
Edit README.md

blowushka committed:
Add couple of tests for CustomAnnotation

blowushka committed:
Complete homework 5

blowushka committed:
HW5 refactoring and add tests

blowushka committed:
Refactoring

blowushka committed:
Edit README.md

blowushka committed:
Add one case for unit-test

blowushka committed:
Major refactoring, add tests, complete CustomRegexMatcher

blowushka committed:
Edit README.md

blowushka committed:
Complete Homework 6

blowushka committed:
Edit README.md

NikolaevArtem commented on src/main/java/homework_2/traffic_light/ExtraTrafficLight.java:
```diff
@@ -0,0 +1,56 @@
+
+public class ExtraTrafficLight {
+
+    public static void run() {
```
Why static?

NikolaevArtem replied:
I thought we deliberately create instance of object, so we don't use static

NikolaevArtem commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,57 @@
+
+import static homework_2.traffic_light.ColorSelector.pickColor;
+
+class TrafficLight {
```
Why not public?

NikolaevArtem commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,57 @@
+
+import static homework_2.traffic_light.ColorSelector.pickColor;
+
+class TrafficLight {
```
![image](https://user-images.githubusercontent.com/58872509/131831580-3447db7d-50e4-43cb-a190-ff14c140de27.png)

NikolaevArtem replied:
Seems you need to debug 
![image](https://user-images.githubusercontent.com/58872509/131831767-459de0e8-80c9-4891-a9ad-36b78c8b1f68.png)

NikolaevArtem replied:
+ table should not have double lines. As in homework chat

NikolaevArtem replied:
+ where are commas?

NikolaevArtem replied:
![image](https://user-images.githubusercontent.com/58872509/131831993-780b0617-6c07-471d-8567-5376fef5ebda.png)

NikolaevArtem commented on src/main/java/homework_2/pyramid_printer/Utils.java:
```diff
@@ -0,0 +1,6 @@
+package homework_2.pyramid_printer;
+
+final class Utils {
```
Opt: seems its not usils, its constants

NikolaevArtem commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,79 @@
+
+    public static void run() {
+        System.out.println("Enter valid table length, width and strategy(even or odd) in format \"length width strategy\" ");
+        getParameters();
```
Wrong naming

blowushka replied:
You mean name for getParameters() method? Should I rename it to getInput() instead, is it ok?

ArtemNikolaev1 replied:
First of all, get methods usually return something. Second, the name of your method does not correlate with what it does. I would really split this method on logic blocks and call them in run() separately. These blocks could be like: getInput(); createTable(); printTable(); smth like this

NikolaevArtem commented on src/main/java/homework_2/random_chars_table/RandomCharsTable.java:
```diff
@@ -0,0 +1,79 @@
+                    }
+                }
+            }
+        } else if (strategy.equals("odd")) {
```
Opt: if not needed

NikolaevArtem commented on src/main/java/homework_2/traffic_light/TrafficLight.java:
```diff
@@ -0,0 +1,57 @@
+    }
+}
+
+//    private static void selectMode(int mode) {
+//        if (mode == 1) {
+//            firstMode();
+//        } else if (mode == 2) {
+//            secondMode();
+//        } else {
+//            System.out.println("You entered wrong mode number.");
+//        }
+//    }
+
+
+//    public static void greeting() {
+//        System.out.println("Pick mode: 1 - provide seconds, 2 - provide time in format \"hh:mm:ss\": ");
+//        try (Scanner scan = new Scanner(System.in)) {
+//            int mode = scan.nextInt();
+//            selectMode(mode);
+//        } catch (InputMismatchException e) {
+//            System.out.println("Only 1 or 2 allow");
+//        }
+//    }
```
Should be removed

blowushka committed:
Major refactoring

blowushka committed:
Edited method names

blowushka committed:
Complete homework_7

blowushka committed:
Edit README.md

blowushka committed:
Fix wrong naming

blowushka committed:
Edit README.md

blowushka committed:
Course project complete

blowushka committed:
Edit README.md

blowushka committed:
Edit README.md

blowushka committed:
Edit MapProblemsMutableGenerator

blowushka committed:
Minor refactoring

NikolaevArtem commented on src/main/java/course_project/Main.java:
```diff
@@ -0,0 +1,8 @@
+
+public class Main {
+
+    public static void main(String[] args)  {
```
Good job! Nice interface, a lot of features (such as automatic ship placement, different game modes), comfortable to play, informative messages. I like the architecture, the code is readable and understandable. You show good knowledge of Java Core (missing Java 8 and some concepts of OOP). 

To make it better: add automatic miss marks placement around a destroyed ship. Also, your code has a lot of static. It seems logical that if you don't care about having multiple objects of a class in your app (like Ship), then mark it static, but actually in this case it should be a singleton, and still dynamic. Static is rarely used in Java, mostly for constants and utils. And mostly your methods are void, which is also rare. 

Approved!

NikolaevArtem commented on src/main/java/homework_3/ImmutableClassMan.java:
```diff
@@ -0,0 +1,62 @@
+6. A parameterized constructor should initialize all the fields performing a deep copy (So that data members can’t be modified with object reference).
+ */
+
+public final class ImmutableClassMan {
```
Integer is immutable, as all wrappers. Please have one field which is mutable

NikolaevArtem commented on src/main/java/homework_4/custom_annotation/HelloGenerator.java:
```diff
@@ -0,0 +1,76 @@
+        try {
+            CustomAnnotation customAnnotation = (CustomAnnotation) this.getClass().getConstructor().getDeclaredAnnotations()[0];
+            temp = customAnnotation.name();
+        } catch (NoSuchMethodException e) {
+            temp = "Aristarch";
+            this.age = 50;
```
opt: That's unexpected.. in catch clause you should handle error, not hide it and surely not insert some "magic" data

NikolaevArtem commented on src/main/java/homework_4/custom_annotation/HelloGenerator.java:
```diff
@@ -0,0 +1,76 @@
+            //this.name = name;
+            this.age = age;
+        } catch (NoSuchMethodException e) {
+            name = "Aristarch";
```
opt: here you put value for local var, not for class field

NikolaevArtem commented on src/main/java/homework_4/custom_annotation/HelloGenerator.java:
```diff
@@ -0,0 +1,76 @@
+        seriesUp(3);
+    }
+
+    public static int[] seriesUp(int n) {
+        int[] res = new int[n * (n + 1) / 2];
+        int index = 0;
+
+        for (int i = 1; i <= n; i++) {
+            for (int j = 0; j < i; j++) {
+                res[index + j] = j + 1;
+            }
+            index += i;
+        }
+
+        System.out.println(Arrays.toString(res));
+        return res;
+    }
```
What is the purpose of this?

NikolaevArtem commented on src/main/java/homework_4/custom_file_reader/CustomFileReader.java:
```diff
@@ -0,0 +1,54 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class CustomFileReader {
```
Good!

NikolaevArtem commented on src/main/java/homework_6/map_problems_generator/MapProblemsGenerator.java:
```diff
@@ -0,0 +1,34 @@
+        MapProblemsMutableGenerator obj1 = new MapProblemsMutableGenerator(125);
+        MapProblemsMutableGenerator obj2 = new MapProblemsMutableGenerator(126);
+
+        mapForUnreachableObjects.put(obj1, "field");
```
Doesn't seem to work 
![image](https://user-images.githubusercontent.com/58872509/134303172-debde803-b0ca-450f-a07c-5e7990c486f5.png)

NikolaevArtem reviewed (changes requested):
Good job! Except for minor fixes in HW_3 and HW_6, homework is done! Today on the lecture we will discuss these homework, as somehow many students have troubles with these

blowushka committed:
Delete seriesUp method
FIX HW_6

blowushka committed:
FIX HW_3
